### PR TITLE
Adding tests for issue #77, closes #77

### LINKED ISSUE #77: Add a test to confirm that all arrays can be empty or null
We need to ensure that we support arrays that are:

- [ ]  empty & present
- [ ] null & present
- [ ] not present

Without unintended side effects (like crashing).

### PR DESCRIPTION
**Pull request by manderiasian**

### CODE REVIEW

File: src/decode.c [hunk 1]
```diff
@@ -137,9 +137,14 @@ int decode_schedule(size_t len, uint8_t * buf, schedule_t **t) {
         msgpack_unpacked_destroy(&result);
     }
 
-    if (0 != ret_val) {
+    if (0 != ret_val || (NULL == s->macs) || (NULL == s->weekly && NULL == s->absolute))
```
Comment (cableramki) [inline src/decode.c]: Does this change make `{}` an error? If so why? It should be allowed as a simple easy means of allowing to specify an empty schedule.

Comment (cableramki) [reply src/decode.c]: If possible to differentiate between `{}` and `{ <invalid entry> }`, it should be handled. 

`{}` may not conform to any schedule, but practically, it is a very simple way to specify empty schedule and should be supported.

Comment (manderiasian) [reply src/decode.c]: This is not the correct place for this discussion, you need to use Slack with Wes and whoever is on the UI side. In general this way of clearing the schedule is a terrible practice ;-) 
The best way would be to add a key/value for a specific action, the decoder can't assume anything, it is the scheduler that has the logic, and an empty file is not the way to go.

Comment (schmidtw) [reply src/decode.c]: The right way to remove a schedule is with a call to `Delete`, not trying to set an empty schedule.
